magefiles/main.go

```diff
@@ -20,21 +21,80 @@ func BootstrapTools() error {
 		Tools []string
 	}
 
-	tools := &ToolsList{}
-	err := readYaml("tools.yaml", tools)
+	requiredTools := &ToolsList{}
+	if err := readYaml("tools.yaml", requiredTools); err != nil {
+		return err
```

Comment:
Should this and the following return err fmt.Printf the message? The standard in this file seems inconsistent, but I'd imagine I'd like to see a specific message if in fact something went wrong. Not a big deal, but you could be saving your future self some heartache. :)

Reply:
returning an error here is actually raising an error to the main process that executed the function. that means that the main process will actually handle and log the error. does that answer your question?